plugins/tracer.py

```diff
@@ -64,50 +67,57 @@ def transaction(self):
 
-        args = ["tracer", "-n"] + list(installed | erased)
-        process = subprocess.Popen(
-            args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-        out, err = process.communicate()
-        _print_output(out, err)
+        try:
+            tracer = TracerFacade(packages)
+            tracer.render()
 
-        if len(out) != 0:
-            print("\n" + _("For more information run:"))
-            print("    sudo tracer -iat " + str(self.timestamp))
+            if len(tracer.apps) != 0:
+                print("\n" + _("For more information run:"))
+                print("    sudo tracer -iat " + str(self.timestamp))
+
+        except Exception:
+            render_error(traceback.format_exc())
 
 
 class TracerCommand(dnf.cli.Command):
     """DNF tracer plugin"""
     aliases = ["tracer"]
 
-    @staticmethod
-    def set_argparser(parser):
-        parser.add_argument("args", nargs="*")
-
     def run(self):
         """Called after running `dnf tracer ...`"""
-        args = ["tracer"] + self.opts.args
-        process = subprocess.Popen(
-            args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-        out, err = process.communicate()
-        _print_output(out, err)
-
-
-def _print_output(out, err):
-    if len(err) != 0:
-        print("Tracer:")
-        print("  " + _("Program 'tracer' crashed with following error:") + "\n")
-        print(err)
-        print(_("Please visit https://github.com/FrostyX/tracer/issues "
-                "and submit the issue. Thank you"))
-        print(_("We apologize for any inconvenience"))
-        return
-
-    if len(out) == 0:
-        print(_("You should restart:"))
-        print("  " + _("Nothing needs to be restarted"))
-        return
-
-    # Last value is blank line
-    for line in out.decode("utf8").split("\n")[:-1]:
-        print(line)
+        print("This plugin calls Tracer after every successful transaction.\n"
+              "If you want to check outdated applications manually, please run standalone `tracer` command.")
+
+
+class TracerFacade(object):
+    def __init__(self, packages, args=None):
+        self.apps = self.get_apps(packages)
+        self.args = args
+
+    def get_apps(self, packages):
+        query = Query()
+        return query.from_packages(packages).now().affected_applications().get()
+
+    def render(self):
+        if len(self.apps) == 0:
+            print(_("You should restart:"))
+            print("  " + _("Nothing needs to be restarted"))
+            return
+
+        # @TODO It is not in the Tracer API yet
+        args = self.args if self.args else dnf.util.Bunch(all=False, quiet=False)
+        view = DefaultView()
+        view.assign("applications", self.apps)
+        view.assign("args", args)
+        return view.render()
+
+
+def render_error(err):
+    print("Tracer:")
+    print("  " + _("Program 'tracer' crashed with following error:") + "\n")
```

Comment:
It doesn't make sense to say `Program 'tracer' crashed` when it is not actually using `/usr/bin/tracer`.